rust_src/src/buffers.rs

```diff
@@ -113,6 +114,28 @@ const fn buf_bytes_max() -> ptrdiff_t {
 }
 pub const BUF_BYTES_MAX: ptrdiff_t = buf_bytes_max();
 
+// When converting bytes from/to chars, we look through the list of
+// markers to try and find a good starting point (since markers keep
+// track of both bytepos and charpos at the same time).
+// But if there are many markers, it can take too much time to find a "good"
+// marker from which to start.  Worse yet: if it takes a long time and we end
+// up finding a nearby markers, we won't add a new marker to cache this
```

Comment:
"finding a nearby markers". This should either be "finding nearby markers" or "finding a nearby marker".